### PR TITLE
Fix share by updating the react native nagivation patches

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Fix share by updating the react native navigation library and patches from main.

#### Ticket Link
NONE

#### Release Note
```release-note
Fix crash that happened when sharing a file
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     "react-native-linear-gradient": "2.8.3",
     "react-native-localize": "3.1.0",
     "react-native-math-view": "3.9.5",
-    "react-native-navigation": "7.39.1",
+    "react-native-navigation": "7.40.0",
```
Comment (enahum) [inline package.json:81]: Is this version the same as in main?

**Review by enahum — CHANGES_REQUESTED** [review]
Podfile.lock needs updating

File: package.json (hunk 1 above)
Comment (larkox) [reply package.json:81]: Yes, it is. https://github.com/mattermost/mattermost-mobile/blob/c6725ab6d3506f0a5216d8dec85fcf83b6f5bb6f/package.json#L83

**Review by enahum — APPROVED** [review]

**Review by rahimrahman — APPROVED** [review]
I did a bit of spot checking on the patch file and seems 👍🏽

**Comment by larkox** [comment]
@yasserfaraazkhan Not sure if you want to test this here, or wait for release testing.

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan Not sure if you want to test this here, or wait for release testing.

I'll quickly check once on PR 🤗

**Review by yasserfaraazkhan — APPROVED** [review]
Able to share files when app is not in the background.

**Pull request merged**